src/diffpy/utils/scattering_objects/diffraction_object.py

```diff
@@ -0,0 +1,492 @@
+import datetime
+from copy import deepcopy
+
+import numpy as np
+
+from diffpy.utils.tools import get_package_info
+
+QQUANTITIES = ["q"]
+ANGLEQUANTITIES = ["angle", "tth", "twotheta", "2theta"]
+DQUANTITIES = ["d", "dspace"]
+XQUANTITIES = ANGLEQUANTITIES + DQUANTITIES + QQUANTITIES
+XUNITS = ["degrees", "radians", "rad", "deg", "inv_angs", "inv_nm", "nm-1", "A-1"]
+
+x_grid_emsg = (
+    "objects are not on the same x-grid. You may add them using the self.add method "
+    "and specifying how to handle the mismatch."
+)
+
+
+class DiffractionObject:
+    """
+    A class to manipulate diffraction data, supporting basic arithmetic operations and
+    conversions between different diffraction data metrics such as Q, two-theta, and d-spacing.
+
+    Parameters
+    ----------
+    name : str, optional
+        Identifier for the diffraction object. Defaults to an empty string.
+    wavelength : float, optional
+        Wavelength of the radiation used in the diffraction experiment. Defaults to None.
+
+    Attributes
+    ----------
+    name : str
+        Name or identifier for the diffraction experiment.
+    wavelength : float
+        Wavelength of the incoming x-rays.
+    scat_quantity : str
+        Description of the type of scattering data stored.
+    on_q : list of numpy.ndarray
+        A list containing two numpy arrays: [Q values, Intensities].
```

Comment:
The "the" constructions helps with these docstrings here, because we need a short statement of what this is, which is missing.  In this case it could be "The diffraction intensities on a grid of Q" followed by something along the ines you already put (i.e., high level view followed by pertinent details).